tidy3d/components/eme/data/dataset.py

```diff
@@ -0,0 +1,95 @@
+"""EME dataset"""
+from __future__ import annotations
+
+
+import pydantic.v1 as pd
+
+from ...data.dataset import Dataset, ElectromagneticFieldDataset
+from ...data.data_array import EMEScalarFieldDataArray, EMESMatrixDataArray
+
+
+class EMESMatrixDataset(Dataset):
+    """Dataset storing S matrix."""
+
+    S11: EMESMatrixDataArray = pd.Field(
+        ...,
+        title="S11 matrix",
+        description="S matrix relating output modes at port 1 to input modes at port 1.",
+    )
+    S12: EMESMatrixDataArray = pd.Field(
+        ...,
+        title="S12 matrix",
+        description="S matrix relating output modes at port 1 to input modes at port 2.",
+    )
+    S21: EMESMatrixDataArray = pd.Field(
+        ...,
+        title="S21 matrix",
+        description="S matrix relating output modes at port 2 to input modes at port 1.",
+    )
+    S22: EMESMatrixDataArray = pd.Field(
+        ...,
+        title="S22 matrix",
+        description="S matrix relating output modes at port 2 to input modes at port 2.",
+    )
+
+
+class EMECoefficientDataset(Dataset):
```

Comment:
Same data structure as EMESMatrixDataset, but the components have different meaning, so better to have it as a different class.